### PR TITLE
Backport HTTP parser upgrade from master to 4.x

### PR DESCRIPTION
**Pull request by ixti**

### CODE REVIEW

**Comment by ixti** [comment]
If there are no objections - I will merge this to 4-x-stable branch and will release 4.2.0 shortly after.

File: README.md [hunk 1]
```diff
@@ -4,13 +4,13 @@
 [![Build Status](https://secure.travis-ci.org/httprb/http.svg?branch=4-x-stable)](https://travis-ci.org/httprb/http)
 [![Code Climate](https://codeclimate.com/github/httprb/http.svg?branch=4-x-stable)](https://codeclimate.com/github/httprb/http)
 [![Coverage Status](https://coveralls.io/repos/httprb/http/badge.svg?branch=4-x-stable)](https://coveralls.io/r/httprb/http)
-[![MIT licensed](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/httprb/http/blob/master/LICENSE.txt)
+[![MIT licensed](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/httprb/http/blob/4-x-stable/LICENSE.txt)
 
 [Documentation]
 
-_NOTE: This is the 4.x **stable** branch.  For the 3.x **stable** branch, please see:_
+_NOTE: This is the 5.x **development** branch.  For the 4.x **stable** branch, please see:_
```
Comment (tarcieri) [inline README.md]: This seems amiss...

**Comment by tarcieri** [comment]
@ixti looks like a few unintentional documentation merges, otherwise LGTM

**Comment by ixti** [comment]
> @ixti looks like a few unintentional documentation merges, otherwise LGTM

Oh, right! Thank you!

File: README.md (hunk 1 above)
Comment (ixti) [reply README.md]: Fixed and force-pushed.

**Comment by ixti** [comment]
@tarcieri please take another look (I've left some README changes, which I believe are OK to be left).

**Comment by zanker** [comment]
Oh awesome, ty!

This looks good to me, but will let Tony stamp it since he took a look initially.

**Comment by ixti** [comment]
Released as v4.2.0

**Pull request merged**